src/main/java/org/spongepowered/common/data/type/SpongeSpawnType.java

```diff
@@ -24,13 +24,28 @@
  */
 package org.spongepowered.common.data.type;
 
+import com.google.common.base.CaseFormat;
 import org.spongepowered.api.event.cause.entity.spawn.SpawnType;
 import org.spongepowered.common.SpongeCatalogType;
 
+import java.util.Locale;
+
 public class SpongeSpawnType extends SpongeCatalogType implements SpawnType {
 
+    private final String name;
+
+    public SpongeSpawnType(String id, String name) {
+        super("sponge:" + CaseFormat.UPPER_CAMEL.to(CaseFormat.LOWER_UNDERSCORE, id));
+        this.name = name;
+    }
+
     public SpongeSpawnType(String id) {
-        super(id);
+        super("sponge:" + CaseFormat.UPPER_CAMEL.to(CaseFormat.LOWER_UNDERSCORE, id));
+        this.name = id;
     }
 
+    @Override
+    public String getName() {
+        return super.getName();
```

Comment:
Why do we return the parents name instead of the name we store a few lines above?